src/SymArrayDmap.chpl

```diff
@@ -92,27 +92,39 @@ module SymArrayDmap {
     }
 
     proc makeDistArray(D: domain(?), type etype) throws
-      where D.rank == 1
+      where MyDmap != Dmap.defaultRectangular && D.isDefaultRectangular()
+    {
+      return makeDistArray(makeDistDom(D), etype);
+    }
+
+    proc makeDistArray(D: domain(?), type etype) throws
+      where D.rank == 1 && (MyDmap == Dmap.defaultRectangular || !D.isDefaultRectangular())
     {
       var res = D.tryCreateArray(etype);
       return res;
     }
 
     proc makeDistArray(D: domain(?), type etype) throws
-      where D.rank > 1
+      where D.rank > 1 && (MyDmap == Dmap.defaultRectangular || !D.isDefaultRectangular())
     {
       var res: [D] etype;
       return res;
     }
 
+    proc makeDistArray(D: domain(?), initExpr) throws
+      where MyDmap != Dmap.defaultRectangular && D.isDefaultRectangular()
+    {
+      return makeDistArray(makeDistDom(D), initExpr);
+    }
+
     proc makeDistArray(D: domain(?), initExpr: ?t) throws
-      where D.rank == 1
+      where D.rank == 1 && (MyDmap == Dmap.defaultRectangular || !D.isDefaultRectangular())
     {
       return D.tryCreateArray(t, initExpr);
     }
 
     proc makeDistArray(D: domain(?), initExpr: ?t) throws
-      where D.rank > 1
+      where D.rank > 1 && (MyDmap == Dmap.defaultRectangular || !D.isDefaultRectangular())
```

Comment:
Note that this is the same case as above and I think it is obvious that whatever we decide on, the behavior should be consistent between these two.